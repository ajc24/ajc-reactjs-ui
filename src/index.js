export { default as Footer } from './ui-elements/Footer';
export { default as Header } from './ui-elements/Header';
export { default as HeaderLogo } from './ui-elements/HeaderLogo';
export { default as HeaderTitleAndSubtitle } from './ui-elements/HeaderTitleAndSubtitle';
export { default as MainContentPageTitle } from './ui-elements/MainContentPageTitle';
export { default as MainContentPanel } from './ui-elements/MainContentPanel';
export { default as MainContentText } from './ui-elements/MainContentText';
export { default as Menu } from './ui-elements/Menu';
export { default as Screenshot } from './ui-elements/Screenshot';
export { default as SmallMenuIcon } from './ui-elements/SmallMenuIcon';
export { default as SmallRightArrowIcon } from './ui-elements/SmallRightArrowIcon';
