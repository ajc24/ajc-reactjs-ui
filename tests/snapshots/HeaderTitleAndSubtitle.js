import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { HeaderTitleAndSubtitle } from '../../src';

describe('HeaderTitleAndSubtitle', () => {
  /* Test data */
  const titleText = 'Web Application Title';
  const subtitleText = 'Web application subtitle / description text goes here';

  describe('Default props and rendering - Header title and subtitle with black text and white outline', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="banner">
          <HeaderTitleAndSubtitle id="test-header-title-and-subtitle-id" titleTextContent={titleText} subtitleTextContent={subtitleText} />
        </div>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Header title and subtitle with white text and black outline', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="banner">
          <HeaderTitleAndSubtitle id="test-header-title-and-subtitle-id" titleTextOutline="black" titleTextColour="white" titleTextContent={titleText} subtitleTextOutline="black" subtitleTextColour="white" subtitleTextContent={subtitleText} />
        </div>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
