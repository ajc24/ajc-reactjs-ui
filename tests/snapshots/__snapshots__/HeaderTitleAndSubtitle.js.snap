// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitleAndSubtitle Default props and rendering - Header title and subtitle with black text and white outline verifies the snapshot for the component 1`] = `
<div
  role="banner"
>
  <div
    className="ajc-header-title-subtitle-container ajc-default"
    id="test-header-title-and-subtitle-id"
  >
    <div
      className="ajc-header-title-subtitle-top"
    >
      <h1
        className="ajc-header-title-subtitle-heading-one ajc-header-text-outline-white ajc-text-black"
      >
        Web Application Title
      </h1>
    </div>
    <div
      className="ajc-header-title-subtitle-bottom"
    >
      <h2
        className="ajc-header-title-subtitle-heading-two ajc-header-text-outline-white ajc-text-black"
      >
        Web application subtitle / description text goes here
      </h2>
    </div>
  </div>
</div>
`;

exports[`HeaderTitleAndSubtitle Transferred props and rendering - Header title and subtitle with white text and black outline verifies the snapshot for the component 1`] = `
<div
  role="banner"
>
  <div
    className="ajc-header-title-subtitle-container ajc-default"
    id="test-header-title-and-subtitle-id"
  >
    <div
      className="ajc-header-title-subtitle-top"
    >
      <h1
        className="ajc-header-title-subtitle-heading-one ajc-header-text-outline-black ajc-text-white"
      >
        Web Application Title
      </h1>
    </div>
    <div
      className="ajc-header-title-subtitle-bottom"
    >
      <h2
        className="ajc-header-title-subtitle-heading-two ajc-header-text-outline-black ajc-text-white"
      >
        Web application subtitle / description text goes here
      </h2>
    </div>
  </div>
</div>
`;
